Tags (#115)
x-govuk/govuk-eleventy-plugin: docs/tags.md

---
layout: sub-navigation
order: 7
title: Adding tags
description: Use tags to categorise posts and make it easier for readers to browse content on your site.
---

> The GOV.UK Eleventy Plugin makes it easy to add tags to your website. Follow these instructions to enable this feature.

Reviewer: Not sure we need to say it’s "easy" (and maybe it’s not _that_ easy given need to create 2 files)?

```suggestion
The GOV.UK Eleventy Plugin lets you use tags to categorise pages. Each post can display its list of tags, which link to a page that lists other posts with the same tag. Follow these instructions to enable this feature.
```

> ## Create a tags page

Reviewer: ```suggestion
## Create a page that lists all tags used
```

> To include a list of all tags used on your website, create a page that uses the `tags` layout:

Reviewer: As below, might be worth suggesting a filename where this content should be copied to?

```suggestion
To include a list of all tags used on your website, create a page that uses the `tags` layout. Add this content to a `tags.md` file:
```

Author: We don’t do this elsewhere… kinda falls into the realms of documenting how Eleventy works.

```yaml
---
eleventyExcludeFromCollections: true
layout: tags
---
```

You can see an [example of a tag index](../layouts/tags) on this website.

> ## Create a tag listing page

Reviewer: I assumed this heading meant the page that listed all tags.

```suggestion
## Create a page for each tag
```

> To generate a page for each tag used on your website, create a page that uses the `tag` layout and paginates `collections.tags` data:

Reviewer: Might be worth giving some guidance on what to call this file? I don't think the filename actually matters that much (as permalink is overridden) but it can be confusing to know where to put these files which generate multiple files based on a collection...

```suggestion
To generate a page for each tag used on your website, create a page that uses the `tag` layout and paginates `collections.tags` data. Add this content to a `tag.md` file:
```

```yaml
---
eleventyExcludeFromCollections: true
layout: tag
pagination:
  data: collections.tags
  size: 1
  alias: tag

> permalink: "/tags/{% raw %}{{ tag | slug }}{% endraw %}/"

Reviewer: Could we do something like this to set the title here?

```suggestion
permalink: "/tags/{% raw %}{{ tag | slug }}{% endraw %}/"
title: Posts tagged ‘{{ tag }}’
```

Author: You would need to use `eleventyComputed.title`, but that blew up (see above comment).

Author: Title for tag pages is now configurable. If you are using breadcrumbs, and want to show them on tag pages, you need to add `eleventyComputed.title`. If not, you can still add a title, otherwise it will default to `Posts tagged ‘[tag]’`.
